includes/Modules/Thank_With_Google.php

```diff
@@ -220,6 +221,43 @@ protected function create_data_request( Data_Request $data ) {
 				return function () {
 					return array();
 				};
+			case 'GET:supporter-wall-sidebars':
+				return function() {
+					$sidebars      = array();
+					$all_sidebars  = wp_get_sidebars_widgets();
+					$block_widgets = get_option( 'widget_block' );
+
+					$pattern = sprintf( '/^%s[0-9]+$/i', preg_quote( Supporter_Wall_Widget::WIDGET_ID . '-', '/' ) );
+					$substr  = sprintf( '"idBase":"%s"', Supporter_Wall_Widget::WIDGET_ID );
+
+					$actual_sidebars_count = 0;
+					foreach ( $all_sidebars as $sidebar_id => $widgets ) {
+						// Skipt the inactive widgets sidebar because it is not an actual sidebar.
+						if ( 'wp_inactive_widgets' === $sidebar_id ) {
+							continue;
+						}
+
+						$actual_sidebars_count++;
+
+						$sidebar = wp_get_sidebar( $sidebar_id );
+						foreach ( $widgets as $widget ) {
+							$block_match = array();
+							if ( preg_match( $pattern, $widget ) ) {
+								$sidebars[ $sidebar_id ] = $sidebar['name'];
+								break;
+							} elseif ( preg_match( '/block-(\d+)/', $widget, $block_match ) && $block_widgets[ $block_match[1] ]['content'] ) {
```

Comment:
Looks like the content here needs to be checked for `$substr`

```suggestion
							} elseif ( preg_match( '/block-(\d+)/', $widget, $block_match ) && strpos( $block_widgets[ $block_match[1] ]['content'], $substr ) ) {
```

Reply:
Yes, correct, thanks for spotting it 👍